apps/webapp/web/static/js/index.js

```diff
@@ -34,17 +34,41 @@ function requireAuth(nextState, replace) {
   }
 }
 
-ReactDOM.render(
-  <Router history={browserHistory}
-    render={applyRouterMiddleware(useRelay)}
-    environment={() => Auth.getEnvironment()}>
-    <Route path="/" component={Hello}/>
+const routes = (
+  <Route path="/" >
+    <IndexRoute component={Hello}/>
     <Route path="/login" component={Login}/>
-    <Route path="/admin" component={AdminLayout} onEnter={requireAuth}>
+    <Route path="/admin" component={AdminLayout} onEnter={requireAuth} >
       <IndexRoute component={Hello}/>
       <Route path="star-wars" component={StarWarsApp} queries={StarWarsQueries}/>
       <Route path="graphiql" component={GraphiQL} />
     </Route>
-  </Router>,
+  </Route>
+)
+
+class Application extends React.Component{
```

Comment:
For now as the code is very simple I won't move this component. But if we get more and more routes and the Application component gets more complex, maybe we could separate the route logic in a Router and the Application component in another file.

In that case the Router could look something like this:
```javascript
class Filter {

  static requireAuth(nextState, replace) {
    if (!Auth.loggedIn()) {
      replace({
        pathname: "/login",
        state: {
          nextPathname: nextState.location.pathname
        }
      })
    }
  }

  static verifySession(nextState, replace) {
    if (Auth.loggedIn()) 
      replace({pathname: "/admin"})
  }
}

export default  (
  <Route path="/">
    <IndexRoute component={Hello}/>
    <Route path="/login" component={Login} onEnter={Filter.verifySession}/>
    <Route path="/admin" component={AdminLayout} onEnter={Filter.requireAuth}>
      <IndexRoute component={Hello}/>
      <Route path="star-wars" component={StarWarsApp} queries={StarWarsQueries}/>
      <Route path="graphiql" component={GraphiQL}/>
    </Route>
  </Route>
)
```